test/src/main/java/org/apache/accumulo/test/WaitForBalanceIT.java

```diff
@@ -63,50 +76,73 @@ public void test() throws Exception {
         scanner.forEach((k, v) -> {});
       }
       c.instanceOperations().waitForBalance();
-      assertTrue(isBalanced(c));
+
+      final List<String> defaultTabletServers =
+          new ArrayList<>(c.instanceOperations().getTabletServers());
+
       final String tableName = getUniqueNames(1)[0];
       NewTableConfiguration ntc = new NewTableConfiguration();
       ntc.withInitialTabletAvailability(TabletAvailability.HOSTED);
-      c.tableOperations().create(tableName, ntc);
-      c.instanceOperations().waitForBalance();
       final SortedSet<Text> partitionKeys = new TreeSet<>();
       for (int i = 0; i < NUM_SPLITS; i++) {
         partitionKeys.add(new Text("" + i));
       }
-      c.tableOperations().addSplits(tableName, partitionKeys);
-      assertFalse(isBalanced(c));
+      ntc.withSplits(partitionKeys);
+      c.tableOperations().create(tableName, ntc);
       c.instanceOperations().waitForBalance();
-      Wait.waitFor(() -> isBalanced(c));
+      Wait.waitFor(() -> isTableBalanced(c, tableName, defaultTabletServers));
+
+      getCluster().getConfig().getClusterServerConfiguration().addTabletServerResourceGroup("TEST",
+          2);
+      getCluster().getClusterControl().start(ServerType.TABLET_SERVER);
+      c.tableOperations().setProperty(tableName, TableLoadBalancer.TABLE_ASSIGNMENT_GROUP_PROPERTY,
+          "TEST");
+      Wait.waitFor(() -> c.instanceOperations().getTabletServers().size() == 4);
+      final List<String> testTabletServers =
+          new ArrayList<>(c.instanceOperations().getTabletServers());
+      assertTrue(testTabletServers.removeAll(defaultTabletServers));
+      // Wait for the TabletManagementIterator to pick up tablet changes
+      Thread.sleep(7000);
```

Comment:
Theoretically we should not need to wait for the TGW to run, waitForBalance should be doing that.   Looked into this and found #4947 as one issue.